dev/test/watch.ts

```diff
@@ -819,6 +824,36 @@ describe('Query watch', () => {
     });
   });
 
+  it('stops attempts after maximum retry attempts', () => {
+    const err = new GrpcError('GRPC Error');
+    err.code = Number(10 /* ABORTED */);
+    return watchHelper.runFailedTest(
+      collQueryJSON(),
+      async () => {
+        let chain = Promise.resolve();
+        // Retry for the maximum of retry attempts.
+        for (let i = 0; i < DEFAULT_MAX_RETRY_ATTEMPTS; i++) {
```

Comment:
You can just use `await` in the for  loop, which should allow you to get rid of the Promise chain.